### DISCUSSION TITLE
How to send json encoded data to a process stdin ?

### DISCUSSION THREAD

**Discussion by jooola** [savonet/liquidsoap#2570] [Q&A]
I would like to send json encoded data (from liquidsoap) to a process stdin (external tool), without necessarily going though a shell.

Though I didn't find any way to pass data via stdin, or to run a process without going through a shell.

I could work around this by passing the data thought the process args, but I feel unsure about passing special chars and properly escaping them to prevent unexpected shell behaviors.

How could I do this ?

**Comment by vitoyucepi** [comment]
I can recommend you to use named pipes or files.

If I were you, I will try to organize ipc via the http.
